diff --git a/node_modules/@superset-ui/embedded-sdk/.DS_Store b/node_modules/@superset-ui/embedded-sdk/.DS_Store
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/@superset-ui/embedded-sdk/dist/index.d.ts b/node_modules/@superset-ui/embedded-sdk/dist/index.d.ts
index b8bf5ad..bcdf90c 100644
--- a/node_modules/@superset-ui/embedded-sdk/dist/index.d.ts
+++ b/node_modules/@superset-ui/embedded-sdk/dist/index.d.ts
@@ -27,6 +27,13 @@ export declare type EmbedDashboardParams = {
     dashboardUiConfig?: UiConfigType;
     /** Are we in debug mode? */
     debug?: boolean;
+    /** Custom header for all the xhr requests **/
+    authToken?: string;
+    /** User info */
+    username?: string;
+    isReadOnly?: boolean;
+    /** locale for i18n */
+    locale?: string;
 };
 export declare type Size = {
     width: number;
@@ -41,4 +48,4 @@ export declare function embedDashboard({ id, supersetD
 /**
  * Embeds a Superset dashboard into the page using an iframe.
  */
-export declare function embedDashboard({ id, supersetDomain, mountPoint, fetchGuestToken, dashboardUiConfig, debug }: EmbedDashboardParams): Promise<EmbeddedDashboard>;
+export declare function embedDashboard({ id, supersetDomain, mountPoint, fetchGuestToken, dashboardUiConfig, debug, authToken, username, isReadOnly, locale }: EmbedDashboardParams): Promise<EmbeddedDashboard>;
diff --git a/node_modules/@superset-ui/embedded-sdk/lib/index.js b/node_modules/@superset-ui/embedded-sdk/lib/index.js
index 5c8591d..f86f958 100644
--- a/node_modules/@superset-ui/embedded-sdk/lib/index.js
+++ b/node_modules/@superset-ui/embedded-sdk/lib/index.js
@@ -41,9 +41,15 @@ async function embedDashboard(_ref) {
     mountPoint,
     fetchGuestToken,
     dashboardUiConfig,
-    debug = false
+    debug = false,
+    authToken,
+    username,
+    isReadOnly,
+    locale
   } = _ref;

+  let refreshTokenTimeoutId = -1;
+
   function log() {
     if (debug) {
       for (var _len = arguments.length, info = new Array(_len), _key = 0; _key < _len; _key++) {
@@ -83,6 +89,9 @@ async function embedDashboard(_ref) {
       const filterConfig = dashboardUiConfig?.filters || {};
       const filterConfigKeys = Object.keys(filterConfig);
       const filterConfigUrlParams = filterConfigKeys.length > 0 ? "&" + filterConfigKeys.map(key => _const.DASHBOARD_UI_FILTER_CONFIG_URL_PARAM_KEY[key] + '=' + filterConfig[key]).join('&') : ""; // set up the iframe's sandbox configuration
+      const userNameParam = `&username=${username}`
+      const isReadOnlyParam = `&isReadOnly=${isReadOnly}`
+      const localeParam = locale ? `&locale=${locale}` : ''

       iframe.sandbox.add("allow-same-origin"); // needed for postMessage to work

@@ -120,30 +129,38 @@ async function embedDashboard(_ref) {
           debug
         }));
       });
-      iframe.src = `${supersetDomain}/embedded/${id}${dashboardConfig}${filterConfigUrlParams}`;
+      iframe.src = `${supersetDomain}/embedded/${id}${dashboardConfig}${filterConfigUrlParams}${userNameParam}${isReadOnlyParam}${localeParam}`;
       mountPoint.replaceChildren(iframe);
       log('placed the iframe');
     });
   }

-  const [guestToken, ourPort] = await Promise.all([fetchGuestToken(), mountIframe()]);
+  // const [guestToken, ourPort] = await Promise.all([fetchGuestToken(), mountIframe()]);
+  // iframe was trying to load while guestToken was still not fetched
+  // So awaiting for guestToken before calling mountIframe
+  const guestToken = await fetchGuestToken();
+  log('Fetched guest token');
+  const ourPort = await mountIframe();
   ourPort.emit('guestToken', {
-    guestToken
+    guestToken,
+    authToken
   });
   log('sent guest token');

   async function refreshGuestToken() {
     const newGuestToken = await fetchGuestToken();
     ourPort.emit('guestToken', {
-      guestToken: newGuestToken
+      guestToken: newGuestToken,
+      authToken
     });
-    setTimeout(refreshGuestToken, (0, _guestTokenRefresh.getGuestTokenRefreshTiming)(newGuestToken));
+    refreshTokenTimeoutId = setTimeout(refreshGuestToken, (0, _guestTokenRefresh.getGuestTokenRefreshTiming)(newGuestToken));
   }

-  setTimeout(refreshGuestToken, (0, _guestTokenRefresh.getGuestTokenRefreshTiming)(guestToken));
+  refreshTokenTimeoutId = setTimeout(refreshGuestToken, (0, _guestTokenRefresh.getGuestTokenRefreshTiming)(guestToken));

   function unmount() {
     log('unmounting');
+    clearTimeout(refreshTokenTimeoutId);
     mountPoint.replaceChildren();
   }

